app/models/job_queue.rb

```diff
@@ -76,13 +90,23 @@ def kill(id)
     @threads[id]&.kill
   end
 
+  def grouped_queue
+    grouped = Hash.new { |h, q| h[q] = [] }
+    @queue.each do |i|
+      grouped[i[:queue]] << i[:job_execution]
+    end
+
+    grouped
+  end
+
   private
 
   def initialize
-    @queue = Hash.new { |h, q| h[q] = [] }
+    @queue = []
     @lock = Mutex.new
     @executing = {}
     @threads = {}
+    JobQueue.concurrency = ENV['MAX_CONCURRENT_JOBS'].to_i
```

Comment:
this looks wrong ... add a delegate and then `self.concurrency = ` ?

Reply:
Sorry, but I'm confused by this comment. I did some quick research and it looks like delegate in Ruby is a way to have a method in one class reference a method from another class. In this case I don't see understand what what what would be delegated to who?

Reply:
what is strange here is that initializing a class sets global state

I think we can simplify this to:

```
def concurrency
  ENV['MAX_CONCURRENT_JOBS'].to_i
end
```

and then in the test you can either do `job_queue.expects(:concurrency).returns(1)` or 

```
with_env MAX_CONCURRENT_JOBS: '1' do
 ... tests ...
end
```